app/src/main/java/com/example/levi9application/viewModels/CocktailViewModel.kt

```diff
@@ -18,21 +19,29 @@ import javax.inject.Inject
 
 @HiltViewModel
 class CocktailViewModel
-@Inject constructor(private val cocktailRepo: CocktailRepo, cocktailDataRepo: CocktailDataRepo) :
+@Inject constructor(private val cocktailRepo: CocktailRepo, private val filterRepo: FilterRepo, cocktailDataRepo: CocktailDataRepo) :
     ViewModel() {
 
     private var job: Job? = null
+    private var job2: Job? = null
```

Comment:
We do not need a job for fetching data based on filter. So we should remove the second job.

Reply:
Do we really need a job inside getFilteredCocktails method at all? Job is used to make a debounce functionality when we search. With getFilteredCocktails method we do not need debouncing at all.